darts/dataprocessing/transformers/base_data_transformer.py

```diff
@@ -582,3 +582,13 @@ def __str__(self):
 
     def __repr__(self):
         return self.__str__()
+
+    def __eq__(self, other) -> bool:
```

Comment:
Could you add some comments here, it's difficult to understand what the equality here means.

for example, when I expected the below test to run fine, but it fails:
```
from darts.datasets import AirPassengersDataset
from darts.dataprocessing.transformers import Scaler

series = AirPassengersDataset().load()
sc1 = Scaler()
sc2 = Scaler()
sc1.fit_transform(series)
sc2.fit_transform(series)

assert sc1 == sc2
```

Reply:
The equality is now comparing the `_fitted_params` entries: either directly np.ndarrays or all the np.ndarray attributes of the object (sklearn scalers)